src/protocols/bdx/BdxTransferProxyDiagnosticLog.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "BdxTransferProxyDiagnosticLog.h"
+
+#include <lib/core/CHIPSafeCasts.h>
+#include <lib/support/CHIPMemString.h>
+#include <platform/LockTracker.h>
+
+namespace chip {
+namespace bdx {
+
+CHIP_ERROR BDXTransferProxyDiagnosticLog::Init(TransferSession * transferSession)
+{
+    VerifyOrReturnError(nullptr != transferSession, CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(nullptr == mTransfer, CHIP_ERROR_INCORRECT_STATE);
+
+    uint16_t fileDesignatorLength = 0;
+    auto fileDesignator           = transferSession->GetFileDesignator(fileDesignatorLength);
+
+    VerifyOrReturnError(fileDesignatorLength > 0, CHIP_ERROR_INVALID_STRING_LENGTH);
+    VerifyOrReturnError(fileDesignatorLength <= ArraySize(mFileDesignator), CHIP_ERROR_INVALID_STRING_LENGTH);
+
+    mTransfer          = transferSession;
+    mFileDesignatorLen = static_cast<uint8_t>(fileDesignatorLength);
+    strncpy(mFileDesignator, Uint8::to_const_char(fileDesignator), fileDesignatorLength);
```

Comment:
Just make this a memcpy.  None of these things are null-terminated or should be null-terminated, so strncpy is not the right thing.

Reply:
Fixed.